rayon-core/src/lib.rs

```diff
@@ -532,6 +537,34 @@ impl<S> ThreadPoolBuilder<S> {
         self
     }
 
+    /// Use the current thread as one of the threads in the pool.
+    ///
+    /// The current thread is guaranteed to be at index 0, and given the thread is not managed by
+    /// rayon, so spawn and exit handlers do not run for that thread.
+    ///
+    /// Note that the current thread won't run the main work-stealing loop, so jobs spawned into
+    /// the thread-pool will generally not be picked up automatically by this thread unless you
+    /// yield to rayon in some way, like via [`yield_now()`], [`yield_local()`], [`scope()`], or
+    /// [`clean_up_use_current_thread()`].
+    ///
+    /// # Panics
+    ///
+    /// This function won't panic itself, but [`ThreadPoolBuilder::build()`] will panic if you've
+    /// called this function and the current thread is already part of another [`ThreadPool`].
```

Comment:
Maybe we _should_ panic opportunistically? I guess you could call this and then never build, but what's the point of that...

An error might be appropriate too, but that doesn't fit the builder pattern well.

Reply:
I'm not a fan of this, both because it complicates the code and because maybe you create a builder in one thread but call `build()` in another? Though I haven't checked if `ThreadPoolBuilder` is `Send`.

Reply:
It's not `Send` because of unconstrained `dyn Fn` fields, like `get_thread_name`.

Reply:
Alright, let me know if you want me to panic before-hand then (and add a static assert that ThreadPoolBuilder isn't Send if you want).

But given we also need the check in build(), I don't think it's worth it necessarily...

Reply:
@cuviper ping about the above? Happy to make this change (or any other) if you want.

Reply:
Eh, this is ok. The cleanup strategy is the other lingering question.